blog/_posts/2021-04-08-scala-3-in-sbt.md

```diff
@@ -0,0 +1,223 @@
+
+```scala
+// build.sbt
+crossScalaVersions := Seq(“2.13.5”, “3.0.0-RC2”)
+```
+
+It is likely that you need to upgrade some dependencies, to change some compiler options or to remove some compiler plugins. Read the [Scala 3 migration tutorial](https://scalacenter.github.io/scala-3-migration-guide/docs/tutorials/sbt-migration.html) to learn more about those steps.
+
+## A new compatibility era
+
+Scala 3 introduces TASTy, an intermediate representation of Scala programs, that is designed for stability during the lifetime of Scala 3.
+The current state of TASTy makes us confident that all Scala 3 minor versions are going to be **backward binary compatible**.
+
+In other words, your Scala 3.x project can depend on libraries compiled with Scala 3.y, where y is lower or equal to x.
+
+This is materialized in sbt by the `scalaBinaryVersion` setting which is set to “3” on all Scala 3 versions (starting from 3.0.0).
+
+Let's take an example:
+
+```scala
+// build.sbt
+scalaVersion := "3.0.0"
+
+organization := "my-org"
+name := "my-project"
+version := "1.0.0"
+
+libraryDependency += "org.typelevel" %% "cats-core" % "x.y.z"
+```
+
+We can print the `scalaBinaryVersion`:
+
+```shell
+sbt:my-project> show scalaBinaryVersion
+[info] 3
+```
+
+As a consequence, the `cats-core` dependency will be resolved to `cats-core_3`.
+
+If I bump the `scalaVersion` to 3.1.0, then the exact same `cats-core_3` artifact will be used and it will still be compatible.
+
+Suppose now that I publish the project as a library, it will be published as `my-project_3` and be available for all projects whose Scala version is greater than 3.1.0.
+
+This is good news for Scala 3 library maintainers, who will not be pressured to release for the latest Scala 3.x version.
+
+For application developers it allows you to use the most recent Scala 3.x version, without worrying about the availability of your dependencies. 
+
+## A smooth transition
+
+While the state described above is an undeniable improvement, it does not make the transition from Scala 2.13 to Scala 3 easier.
+Yet we claim this transition will be smoother than ever thanks to the interoperability between the two versions of the language.
+
+### Disclaimer for library maintainers
+
+Using the interoperability between Scala 2.13 and Scala 3 in a published library is generally not safe for your end-users. 
+
+Unless you know exactly what you are doing, it is discouraged to publish a Scala 3 library that depends on a Scala 2.13 library (the scala-library being excluded) or vice versa.
+
+The reason is to prevent library users from ending up with two conflicting versions `x_2.13` and `x_3` of the same x library in their classpath, this problem being unsolvable in some cases.
+
+### Using Scala 2.13 libraries in Scala 3
+
+The Scala 3 compiler is able to type check Scala 2.13 types and to resolve Scala 2.13 implicits.
+It means you can safely use Scala 2.13 libraries in a Scala 3 application.
```

Comment:
This paragraph makes me think I can use _any_ 2.13 library in a Scala 3 application, but I have to wait for the last paragraph to see that it applies only to libraries that don’t use macros. I think this is an important point, I wonder if it would be possible to warn about that in this paragraph.

```suggestion
The Scala 3 compiler is able to type check Scala 2.13 types and to resolve Scala 2.13 implicits. However, it can’t expand Scala 2.13 macros.
It means you can safely use Scala 2.13 libraries in a Scala 3 application, but you can’t use Scala 2.13 macros in a Scala 3 application.
```